### DISCUSSION TITLE
用词调整讨论 Word

### DISCUSSION THREAD

**Discussion by sena-nana** [sena-nana/basedpyright#3] [Ideas]
如题

**Comment by NCBM** [comment]
**incompatible** 一词用于类型诊断总体无法等同的报告。

例：`Expression of type "{exprType}" is incompatible with return type "{returnType}"`

此时的 **incompatible** 个人提供了以下几种译法：

- 不兼容（的）——字面翻译
- 不相容（的）
- 不匹配（的）

在此处征求词语选用意见。

**Reply by sena-nana** [reply]
从个人使用的感觉来看，不匹配是最直观的。但也有逆变协变那种情况，类型与返回类型并不相等，我不确定这种情况下用匹配是否还合适

**Comment by NCBM** [comment]
**pattern** 一词多指用于模式匹配的形式格式。

例：`Irrefutable pattern is allowed only for the last case statement`
（注：这里的 **Irrefutable pattern** 指的就是 `case _`）

此时的 **pattern** 个人提供了以下几种译法：

- 模式——字面翻译
- 形式

不过个人感觉好像没什么充分理由用后者替换前者。

**Reply by sena-nana** [reply]
确实

**Comment by NCBM** [comment]
语言文件中有下列条目（已省略无关条目）：

```json
{
    "Diagnostic": {
        "typedDictClosedExtras": "Base class \"{name}\" is a closed TypedDict; extra items must be type \"{type}\"",
        "typedDictClosedNoExtras": "Base class \"{name}\" is a closed TypedDict; extra items are not allowed"
    },
    "DiagnosticAddendum": {
        "typedDictClosedExtraNotAllowed": "Cannot add item \"{name}\"",
        "typedDictClosedExtraTypeMismatch": "Cannot add item \"{name}\" with type \"{type}\"",
        "typedDictClosedFieldNotRequired": "Cannot add item \"{name}\" because it must be NotRequired"
    }
}
```

经查阅得知这些条目与 [PEP 728](https://peps.python.org/pep-0728/) 有关。目前尚未确定 `closed TypedDict` 的合适替换翻译。

**Reply by sena-nana** [reply]
目前py文档里应该还没有对应的翻译？不过我感觉以后会被翻译成 `封闭的TypedDict` 的概率会比较大（
